src/zenml/stack/stack.py

```diff
@@ -123,6 +121,55 @@ def __init__(
         self._annotator = annotator
         self._data_validator = data_validator
 
+        requires_image_builder = (
+            orchestrator.flavor != "local"
+            or step_operator
+            or (model_deployer and model_deployer.flavor != "mlflow")
```

Comment:
True. Do you think `model_deployer.flavor in ["seldon", "kserve", "bento"]` is better?

Reply:
Yeah I would prefer that. Generally speaking, that doesn't fully solve the issue for custom stack components either though. 

Ideally, I think it should be configurable somewhere in the stack component itself whether it needs the image builder or not. Perhaps we could add a property `@requires_image_builder(self) -> bool` similar to how we have `is_local` right now?

Reply:
Discussion: this code is only used for backwards-compatibility to not break existing stacks. Fur custom components, adding an unnecessary image builder is better than not having one when it would be needed, so requiring image builders for as many components as possible makes sense.